text/0031-expanded_remove.md

```diff
@@ -0,0 +1,236 @@
+
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The remove command unpins blocks AND alters the logbook. Removing a read-only dataset drops logbook data entirely. Removing from a writable dataset adds an operation to the dataset log marking the number of versions removed.
+
+All the previously stated local delete behaviors stay the same, with a few exceptions listed below.
+But in addition to those behaviors, we have remote behaviors as well:
+```
+$ qri remove me/dataset --remote registry
+```
+Unpins all blocks & removes the log on the registry because the registry has a read-only copy. Does nothing locally.
+
+Attempting to drop revisions from a read-only dataset replica is an error:
+```
+$ qri remove me/dataset --remote registry --revisions 1
+ERROR
+```
+
+This you can't manipulate the logbook of a dataset on a remote. Instead you would first edit locally & push:
+```
+$ qri remove me/dataset --revisions 1
+$ qri push me/dataset
+```
+
+remove works without any flags on read-only datasets:
+```
+$ qri remove not_me/dataset
+```
+unpins the blocks and then removes the log of a dataset that is not in your namespace. Because we have to remove the log entirely, we cannot restore a dataset that was removed but not in your namespace. Instead restore by re-pulling.
+
+using revisions with a read-only dataset will error
+```
+$ qri remove not_me/dataset --revisions 1
+ERROR
+```
+should error: you can't manipulate the logbook of a read-only dataset.
+
+If you alter your logbook locally using qri remove, you can update the remote log by using qri push.
+
+
+### Use the logbook
+Delete actions should examine the history of a logbook to present warnings to a user that help them "clean up" when deleteing stuff. As an example: `$ qri remove --all` on a dataset should look for `push` operations and warn user they should `drop` from a list of remotes.
+
+### rename the `--drop` flag on `qri save`
+The current master branch has a `--drop` flag on the save command for "dropping components of a dataset". This RFC repurposes the `--drop` language to be about datasets you don't own, so this flag should be renamed to `--remove`. We shouldn't use `--delete`, because the word delete is now reserved for editing history.
+
+### Corner Cases
+There are a few scenarios we should make a point writing tests for:
+* Running amend on a dataset with no commits (no history) is an error
+* a drop request for a dataset name that doesn't exist locally, but _does_ exist on the remote should not error. This implies remotes should prefer local reference resolution
+* Remotes _should_ accept altered InitIDs when an owner Pushes to them. If user creates `b5/population`, pushes, then `delete --all b5/population` locally, then creates a new dataset with the same `b5/population` name, and pushes again, that should work, completely replacing the history
```

Comment:
I would rephrase this as "Remotes should uniquely identify datasets by their initID, rather than their name, when a new dataset is pushed to them. A name can be changed or reused while a initID can not." The important part here is that, while we ensure names are unique, they can change, so they aren't stable ways to identify datasets. The way it is currently phrased inverts the relationship, making it seem like a *name* lasts forever, and the initID for that name is being altered.

Rather, we should think of it like a user signing up for a new website. Though each user name can only be used once, it shouldn't be the primary key that is used internally to refer to their database row. That way the username can be changed later on, or reused if the first user deletes their account.